rstudio-entrypoint.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+# On Linux set rstudio user id to same as id of host if rstudio user id not already the same
+if [ "$HOSTPLATFORM" = "linux" -a "$(id -u rstudio)" != "$HOSTUID" ]; then
+  usermod -u $HOSTUID rstudio
+fi
+
+# Check for 1 .Rproj file
+if [ $(find /workspace -type f -name "*.Rproj" | wc -w) -eq 1 ]; then
+
+  # Copy in Git user.name and user.email from copied local-gitconfig from additionally mounted volume
+  if test -f /home/rstudio/local-gitconfig; then
+    grep -e "\[user\]" -e "name = *" -e "email = *" /home/rstudio/local-gitconfig >> /home/rstudio/.gitconfig
+  fi
+
+  # Avoid Git error: fatal detected dubious ownership of repository if using Git in container
+  # Without this the Git pane fails to open when RStudio project opened
+  echo -e "[safe]\n\tdirectory = \"*\"" >> /home/rstudio/.gitconfig
+
+  # Open RStudio project on opening RStudio Server session
+  echo 'setHook("rstudio.sessionInit", function(newSession) { if (newSession && is.null(rstudioapi::getActiveProject())) rstudioapi::openProject(paste0("/workspace/", list.files(pattern = "Rproj"))) }, action = "append")' >> /home/rstudio/.Rprofile
```

Comment:
Consider moving this to a file checked into the repo, copied to the image, and then moved to /home/rstudio/.Rprofile here if needed.

This will make it easier to understand and potentially change in future.

Reply:
I have moved this to an R file as suggested - which is copied into the image. It's renamed to .Rprofile if an .Rproj is detected.